tensorflow/contrib/learn/python/learn/estimators/estimators_test.py

```diff
@@ -77,6 +78,44 @@ def model_fn(features, labels):
     # labels = transformed_y (99)
     self.assertEqual(99., metrics["label"])
 
+  def testFeatureEngineeringFnWithSameName(self):
+
+    def input_fn():
+      return {
+               "x": constant_op.constant(["9."])
+             }, {
+               "y": constant_op.constant(["99."])
+             }
+
+    def feature_engineering_fn(features, labels):
+      # Github #12205: raise a TypeError if called twice.
+      _ = string_ops.string_split(features["x"])
+      features["x"] = constant_op.constant([9.])
+      label["y"] = constant_op.constant([99.])
+      return features, labels
```

Comment:
Hi, @ispirmustafa . In the `feature_engineering_fn`, it converts string tensor to float tensor. So if it's be called twice, TypeError will be raised because `features["x"]` has been converted to float tensor, which is unsupported by string_op.